microprofile/WFLY-12609_MicroProfile_JWT_1_1.adoc

```diff
@@ -0,0 +1,153 @@
+
+Activation will add module dependencies as required on SmallRye JWT, Elytron JWT, and the MicroProfile JWT API.
+
+Activation will also ensure activation of the SmallRye JWT CDI extension as well as the Elytron JWT CDI extension for Elytron specific overrides.
+
+Finally activation will ensure the activation of the `MP-JWT` authentication mechanism for the deployment.
+
+The combination of SmallRye JWT with Elytron JWT will be the pair of projects responsible for providing the functionality described in the MicorProfile JWT specification combined with tying in the integration to the application server's security framework to ensure aspects such as identity association and establishment are correctly performed.
+
+Once an identity has been established according to the steps described by the MicroProfile JWT specification the remaining security decisions required are covered by the relevant Jakarta EE specifications in use by the various subsystems. 
+
+=== Nice-to-Have Requirements
+
+This section is deliberately empty, the intent of this enhancement is to integrate MicroProfile JWT support to WildFly.  Once the initial integration is in place nice to have functionality can be added under individual enhancements with an appropriately constrained scope.
+
+=== Non-Requirements
+
+This integration will be making use of WildFly Elytron exclusively, no support will be added for the deprecated PicketBox project.  Having said that the underlying security framework will be transparent for deployments developed according to the specification - the framework only becomes apparent if there is a desire to use vendor specific APIs.
+
+This integration will not be supporting the use of pre-configured SecurityDomains, the general motivation to use MicroProfile JWT is to use an authentication mechanism where identities are self described using the tokens exchanged eliminating the need for configured security resources for the secured endpoint(s).  Use cases could be identified where pre-configured resources add an additional benefit but those would be reviewed under their own enhancements.
+
+It is assumed when working with the MicroProfile specifications that we will be supporting a single deployment, this deployment may contain sub-deployments.  The SecurityDomain of each top level deployment will be isolated from the SecurityDomain of any parallel top level deployment.  Advanced WildFly Elytron features such identity propagation across domains / top level deployments will not be supported.  At a later point we could consider an enhancement to offer more control of the SecurityDomains in use but that is out of scope for this enhancement.
```

Comment:
>  It is also possible that MicroProfile JWT is activated for a deployment that it itself a sub-deployment such as a `war` contained within an `ear` or in cases where the `war` contains items to be deployed in another container such as EJBs within the `war`.  In all of these cases the common "Virtual" `SecurityDomain` will be used across all containers handling the deployment - by using a common `SecurityDomain` any identity established in one container will automatically propagate with any calls to the other containers without the need for any manual configuration. 

This example describes successive levels, so it is not in conflict with this non-requirement, right? Or is it regarding other issue?

Reply:
For now it may be worth adding a note to say mixed deployments will not be supported, i.e. an ear that contains a war using mp-jwt in parallel with a war that uses a different mech.

The motivation here is MicroService development not monolith development so we can always pick that up later with the support for the other mechanisms.